File: Problem_set_1/Introductory_problems/p09_Bit_Strings/BitStrings.java
import java.util.Scanner;

public class BitStrings {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        long length = scan.nextLong();
        long result = 1;
        long mod = 1000000007;

        for(int i = 0; i < length; i++){
            result = (result * 2) % mod;
        }

        System.out.println(result);
        scan.close();
    }
}

File: Problem_set_1/Introductory_problems/p09_Bit_Strings/BitStrings.cpp
#include<bits/stdc++.h>

using namespace std;

int main(){
    long long length, result = 1, mod = 1000000007;
    cin >> length;

    for(int i = 0; i < length; i++){
        result = (result * 2) % mod;
    }

    cout << result << endl;

    return 0;
}